### PR TITLE
Improve JSDoc in gd.js

### PR DESCRIPTION
**Pull request by arthuro555**

### CODE REVIEW

File: GDJS/Runtime/gd.js [hunk 1]
```diff
@@ -70,13 +85,20 @@ gdjs.randomFloat = function(max) {
...
  */
 gdjs.randomFloatInRange = function(min, max) {
   return min + gdjs.randomFloat(max - min); // return min if min >= max
 };
 
 /**
  * Get a random number between min and max in steps
+ * @param {number} min The minimum value.
+ * @param {number} max The maximum value.
+ * @param {number} step The step.}
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: step The interval between each value.

File: GDJS/Runtime/gd.js [hunk 2]
```diff
@@ -33,35 +33,50 @@ window.gdjs = {
...
+ * @param {number} min The minimum value.
+ * @param {number} max The maximum value.
+ * @returns {number}
  */
 gdjs.randomInRange = function(min, max) {
   return min + gdjs.random(max - min); // return min if min >= max
 };
 
 /**
  * Get a random float between 0 and max.
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: Let's make this description a bit more complete:

Get a random float in the range 0 to less than max (inclusive of 0, but not max).

File: GDJS/Runtime/gd.js [hunk 3]
```diff
@@ -33,35 +33,50 @@ window.gdjs = {
...
+ * @param {number} max The maximum value.
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: The maximum value (exclusive)

File: GDJS/Runtime/gd.js [hunk 4]
```diff
@@ -70,13 +85,20 @@ gdjs.randomFloat = function(max) {
 
 /**
  * Get a random float between min and max
+ * @param {number} min The minimum value.
+ * @param {number} max The maximum value.
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: The minimum value (inclusive).
The maximum value (exclusive).

File: GDJS/Runtime/gd.js [hunk 5]
```diff
@@ -33,35 +33,50 @@ window.gdjs = {
...
+ * @param {number} b Blue
+ * @returns {number}
  */
 gdjs.rgbToHexNumber = function(r, g, b) {
   return (r << 16) + (g << 8) + b;
 };
 
 /**
  * Get a random integer between 0 and max.
+ * @param {number} max The maximum value.
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: The maximum value (inclusive).

File: GDJS/Runtime/gd.js [hunk 6]
```diff
@@ -33,35 +33,50 @@ window.gdjs = {
...
 gdjs.random = function(max) {
   if (max <= 0) return 0;
   return Math.floor(Math.random() * (max + 1));
 };
 
 /**
- * Get a random integer between min and max
+ * Get a random integer between min and max.
+ * @param {number} min The minimum value.
+ * @param {number} max The maximum value.
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: The minimum value (inclusive).
The maximum value (inclusive).

File: GDJS/Runtime/gd.js [hunk 7]
```diff
@@ -70,13 +85,20 @@ gdjs.randomFloat = function(max) {
...
+ * @returns {number}
  */
 gdjs.randomFloatInRange = function(min, max) {
   return min + gdjs.randomFloat(max - min); // return min if min >= max
 };
 
 /**
  * Get a random number between min and max in steps
+ * @param {number} min The minimum value.
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: The minimum value (inclusive).

File: GDJS/Runtime/gd.js [hunk 8]
```diff
@@ -70,13 +85,20 @@ gdjs.randomFloat = function(max) {
...
+ * @param {number} max The maximum value.
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: The maximum value (inclusive)

File: GDJS/Runtime/gd.js [hunk 9]
```diff
@@ -85,18 +107,30 @@ gdjs.randomWithStep = function(min, max, step) {
...
+ * @param {number} angleInRadians The angle in radians.
+ * @returns {number}
  */
 gdjs.toDegrees = function(angleInRadians) {
   return (angleInRadians * 180) / 3.14159;
 };
 
+/**
+ * A Constructor for a {@link gdjs.RuntimeObject}.
+ * @name ObjectCtor
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: Let's maybe call this RuntimeObjectCtor - or even RuntimeObjectConstructor.

Reasons:
* "Runtime" for consistency with RuntimeObject.
* Object alone is too close to a JS "Object" and could creator confusion.
* "Constructor" => let's use full terms.

File: GDJS/Runtime/gd.js [hunk 10]
```diff
@@ -105,12 +139,21 @@ gdjs.toDegrees = function(angleInRadians) {
...
- * @param Ctor The constructor of the Object.
+ * @param {ObjectCtor} Ctor The constructor of the Object.
  */
 gdjs.registerObject = function(objectTypeName, Ctor) {
   gdjs.objectsTypes.put(objectTypeName, Ctor);
 };
 
+/**
+ * A Constructor for a {@link gdjs.RuntimeBehavior}.
+ * @name BehaviorCtor
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: Same, let's go for RuntimeBehaviorConstructor

File: GDJS/Runtime/gd.js (hunk 2 above)
Comment (4ian) [reply GDJS/Runtime/gd.js]: Ping :)

File: GDJS/Runtime/gd.js [hunk 11]
```diff
@@ -33,35 +33,50 @@ window.gdjs = {
...
+ * @returns {number}
  */
 gdjs.random = function(max) {
   if (max <= 0) return 0;
   return Math.floor(Math.random() * (max + 1));
 };
 
 /**
- * Get a random integer between min and max
+ * Get a random float in the range 0 to less than max (inclusive of 0, but not max).
```
Comment (4ian) [inline GDJS/Runtime/gd.js]: This description is not correct

**Comment by 4ian** [comment]
I need you to be a bit more careful about your changes - it's time consuming for me to do multiple times the review and I will sadly have less time in the future as the community grows :/ 

Always double check your PR by re-reading the code like I would do. Ensure the tests are passing (right now some type checking is failing), triple check your comments. I try to review quickly but I can easily miss some mistakes in comments.

**Comment by arthuro555** [comment]
I am sorry I was in a bit of a hurry and did it too fast and did this stupid mistake :sweat: . I will fix it ASAP.
For the type checks, I am very surprised. It comes from a for loop in runtimeObject.js that doesn't seem related to the changes in gd.js. I am puzzled.

File: GDJS/Runtime/runtimeobject.js [hunk 12]
```diff
@@ -1357,10 +1357,7 @@ gdjs.RuntimeObject.prototype.separateObjectsWithoutForces = function(objectsList
  * @deprecated
  * @param objectsLists Tables of objects
  */
-gdjs.RuntimeObject.prototype.separateObjectsWithForces = function(objectsLists, len) {
-
-    if ( len == undefined ) len = 10;
-
+gdjs.RuntimeObject.prototype.separateObjectsWithForces = function(objectsLists, dummyVar) {
```
Comment (4ian) [inline GDJS/Runtime/runtimeobject.js]: Seems that you're right, the value will be erased anyway.
You can remove `dummyVar`, JavaScript is not crashing if extra arguments are passed and not used. (Typescript would error, but not JS)/

**Comment by 4ian** [comment]
Alright! Looks correct, merging now :) Thanks for working on this 👍

**Pull request merged**